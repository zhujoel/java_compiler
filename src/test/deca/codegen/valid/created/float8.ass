; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 11(GB)
	LOAD code.Object.equals, R0
	STORE R0, 12(GB)
code.Object.equals:
; Main Function
; Beginning of main instructions:
; Test pour savoir si la pile est pleine
	TSTO #4
	BOV pile_pleine
	ADDSP #4
; Declaration des variables
	LOAD #0x1.266666p1, R2
	STORE R2, 13(GB)
	LOAD #0x1.333334p0, R2
	STORE R2, 14(GB)
; Instructions
; *
	LOAD 13(GB), R2
	LOAD 14(GB), R3
	MUL R2, R3
	BOV debordement
	LOAD 13(GB), R1
	WFLOAT
	HALT
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
