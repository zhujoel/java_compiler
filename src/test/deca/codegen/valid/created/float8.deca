{
    float a = 2.3;
    float b = 1.2;
    a * b;
    print(a);
}
