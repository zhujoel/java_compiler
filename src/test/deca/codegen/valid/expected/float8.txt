2.30000e+00
